File: app/Models/Scholarship.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Scholarship extends Model
{
    protected $fillable = [
        'title',
        'slug',
        'country',
        'description',
        'deadline',
        'apply_url',
        'image',
        'status',
        'is_featured',
        'sort_order',
    ];

    protected $casts = [
        'deadline' => 'date',
        'is_featured' => 'boolean',
    ];

    public function isPublished(): bool
    {
        return $this->status === 'published';
    }

    public function isExpired(): bool
    {
        return $this->deadline !== null && $this->deadline->isPast();
    }

    public function imageUrl(): ?string
    {
        return $this->image ? asset('storage/' . $this->image) : null;
    }

    public function scopePublished($query)
    {
        return $query->where('status', 'published');
    }

    public function scopeFeatured($query)
    {
        return $query->where('is_featured', true);
    }

    public function scopeOpen($query)
    {
        return $query->where(function ($q) {
            $q->whereNull('deadline')->orWhere('deadline', '>=', now()->toDateString());
        });
    }
}

File: mhub/lib/models/scholarship.dart
class Scholarship {
  const Scholarship({
    required this.id,
    required this.title,
    required this.slug,
    this.country,
    this.isFeatured = false,
    this.imageUrl,
    this.shortDescription,
    this.description,
    this.deadlineLabel,
    this.isExpired = false,
    this.applyUrl,
  });

  final int id;
  final String title;
  final String slug;
  final String? country;
  final bool isFeatured;
  final String? imageUrl;
  final String? shortDescription;
  final String? description;
  final String? deadlineLabel;
  final bool isExpired;
  final String? applyUrl;

  factory Scholarship.fromJson(Map<String, dynamic> j) => Scholarship(
        id: (j['id'] as num).toInt(),
        title: j['title'] as String? ?? '',
        slug: j['slug'] as String? ?? '',
        country: j['country'] as String?,
        isFeatured: j['is_featured'] == true,
        imageUrl: j['image_url'] as String?,
        shortDescription: j['short_description'] as String?,
        description: j['description'] as String?,
        deadlineLabel: j['deadline_label'] as String?,
        isExpired: j['is_expired'] == true,
        applyUrl: j['apply_url'] as String?,
      );
}
